crates/nu-protocol/src/ast/expr.rs

```diff
@@ -54,6 +46,11 @@ pub enum Expr {
     Garbage,
 }
 
+// This is to document/enforce the size of `Expr` in bytes.
+// We should try to avoid increasing the size of `Expr`,
+// and PRs that do so will have to change the number below so that it's noted in review.
+const _: [(); 40] = [(); std::mem::size_of::<Expr>()];
+
```

Comment:
Thoughts on adding this assert?

Reply:
I think that is a good call, we could also do it for `Value` as that's even hotter.

I don't know if this is only possible since somewhat recent versions but this could also be written as:
```suggestion
// This is to document/enforce the size of `Expr` in bytes.
// We should try to avoid increasing the size of `Expr`,
// and PRs that do so will have to change the number below so that it's noted in review.
const _: () = assert!(std::mem::size_of::<Expr>() == 40);

```

(limited to `assert!` as `assert_eq!` and friends have non `const` functions in them)